util/matrix.hxx

```diff
@@ -602,6 +618,41 @@ void op::dot(const Mat<N,P,T>& m1, const Mat<P,Q,T>& m2, Mat<N,Q,T>& res)
 // }
 
 
+template<uint32_t M, uint32_t N, uint32_t P, uint32_t Q, typename T>
+bool op::insert(const Mat<M,N,T>& m1, const Mat<P,Q,T>& m2, uint32_t i_start, uint32_t j_start, Mat<M,N,T>& res)
+{
```

Comment:
i_start and j_start can be template to static_assert that 
i_start + P <= M 
and 
j_start + Q <= N